File: src/main/java/at/hrechny/predictionsbot/model/Result.java
package at.hrechny.predictionsbot.model;

import lombok.Data;

@Data
public class Result {

  private User user;

  private Integer predictions;

  private Integer predictionsLive;

  private Integer guessed;

  private Integer guessedLive;

  private Integer sum;

  private Integer liveSum;

  public Integer getTotalPredictions() {
    return predictionsLive != null ? predictions + predictionsLive : predictions;
  }

  public Integer getTotalGuessed() {
    return guessedLive != null ? guessed + guessedLive : guessed;
  }

  public Integer getTotalSum() {
    return liveSum != null ? sum + liveSum : sum;
  }

}

File: src/main/kotlin/at/hrechny/predictionsbot/model/Result.kt
package at.hrechny.predictionsbot.model

import io.micronaut.core.annotation.Introspected

@Introspected
class Result {
    var user: User? = null
    var predictions: Int? = null
    var predictionsLive: Int? = null
    var guessed: Int? = null
    var guessedLive: Int? = null
    var sum: Int? = null
    var liveSum: Int? = null

    val totalPredictions: Int?
        get() = if (predictionsLive != null) predictions!! + predictionsLive!! else predictions

    val totalGuessed: Int?
        get() = if (guessedLive != null) guessed!! + guessedLive!! else guessed

    val totalSum: Int?
        get() = if (liveSum != null) sum!! + liveSum!! else sum
}
